### PR TITLE
chore: group dependencies

### PR DESCRIPTION
**Pull request by matifali**
This should help reduce dependabot noise. Probably grouping can be further improved.

### CODE REVIEW

File: .github/dependabot.yaml
Comment (matifali) [inline .github/dependabot.yaml:117]: Alternatively, we can add a group for `storybook` updates too.

Comment (aslilac) [reply .github/dependabot.yaml:121]: If dependabot had a way to run custom scripts we could. The upgrade command often includes migrations to help you move away from deprecated features to their newer counterparts, and it'd be sad to miss out on those.

Comment (matifali) [reply .github/dependabot.yaml:121]: I agree. Sadly currently there isn't a way to do this. So I guess it's better to ignore storybook and do it manually. What do you think?

Comment (aslilac) [reply .github/dependabot.yaml:121]: agreed

**Comment by aslilac** [comment]
I know we had this before, so I'm curious why it was removed. Was it causing any issues?

**Comment by matifali** [comment]
@aslilac we did had them in https://github.com/coder/coder/pull/8453 but were still not happy with amount of PRs dependabot was creating. So we created a single large group of everything in https://github.com/coder/coder/pull/10870. This made the PRs unmergeable and was creating PRs with 67 dependencies update in a single PR.
So again adding some very carefully chosen asne groups should provide an acceptable middle ground.

**Review by aslilac — APPROVED** [review]
thanks for explaining, I like this approach 😄

**Comment by matifali** [comment]
@aslilac Please go ver the packages.json and dependabot.yaml to see if it can be further optimized. 
I am waiting for @coadler approval too.

**Review by coadler — APPROVED** [review]
The `golang.org/x/*` group seems good to me

**Pull request merged**
